kvbc/src/ReplicaImp.cpp

```diff
@@ -674,9 +579,9 @@ bool ReplicaImp::BlockchainAppState::getPrevDigestFromBlock(uint64_t blockId, St
     exit(1);
   }
 
-  const BlockHeader *bh = reinterpret_cast<const BlockHeader *>(result.data());
+  auto parentDigest = block::getParentDigest(result);
```

Comment:
Eventually maybe we can start using proper C++ types like the SHA3 hash in util. Then we won't need memcpy, etc... Definitely do not change that in this PR though. Just thinking out loud.

Reply:
Also, nice cleanup :)